zone/doors.cpp

```diff
@@ -190,6 +190,204 @@ bool Doors::Process()
+		return false;
+	}
+
+	if (sender->IsLockedOutOfInstance(zoneid))
+	{
+
+		// TODO: GetTargetZoneInstanceID / Get raid leader's instance ID if not leader / if leader and not locked out, issue new ID, if not, message
+		Raid* player_raid = sender->GetRaid();
+
+		if (!player_raid)
+		{
+			sender->Message(Chat::Red, "You are unable to enter an instance because you are not a part of a raid.");
+			return false;
+		}
+
+		if (player_raid->MeetsInstancedRaidRequirements() < RuleI(Quarm, AutomatedRaidRotationRaidNonMemberCountRequirement))
+		{
+			sender->Message(Chat::Red, "You are unable to enter an instance because you are not a part of a raid with %i players at or above level %i present total.",
+				RuleI(Quarm, AutomatedRaidRotationRaidNonMemberCountRequirement),
+				RuleI(Quarm, AutomatedRaidRotationRaidGuildLevelRequirement));
+			return false;
+		}
+
+		ourZoneInstanceID = sender->GetTargetZoneInstanceID(zoneid);
+		zoneguildid = ourZoneInstanceID;
+
+		int64 curzoneLockout = zone->GetZoneInstanceLockoutByCharacterAndZone(sender->CharacterID(), zoneid);
+		//TODO: hardcode, retrieve from db later
+		int64 zonelockout = RuleI(Quarm, InstanceMinimumLockoutTime);
+
+		auto cur_time = time(nullptr);
+
+		int64 end_time = cur_time + RuleI(Quarm, InstanceMinimumLockoutTime);
+		if (curzoneLockout <= end_time)
+		{
+			zone->ReplaceZoneInstanceIDCache(sender->CharacterID(), zoneid, zoneguildid, end_time);
+			database.SaveCharacterInstanceLockout(sender->CharacterID(), end_time, zoneid, zoneguildid);
+			CharacterInstanceLockout instanceLockout;
+			memset(&instanceLockout, 0, sizeof(CharacterInstanceLockout));
+			instanceLockout.character_id = sender->CharacterID();
+			instanceLockout.expirydate = end_time;
+			instanceLockout.zone_id = zoneid;
+			instanceLockout.zone_instance_id = zoneguildid;
+			sender->character_instance_lockouts[zoneid] = instanceLockout;
+		}
+
+	}
+	else
+	{
+		// TODO: GetTargetZoneInstanceID / Get raid leader's instance ID if not leader / if leader and not locked out, issue new ID, if not, message
+		Raid* player_raid = sender->GetRaid();
+
+		if (!player_raid)
+		{
+			sender->Message(Chat::Red, "You are unable to enter an instance because you are not a part of a raid.");
+			return false;
+		}
+
+		if (player_raid->MeetsInstancedRaidRequirements() < RuleI(Quarm, AutomatedRaidRotationRaidNonMemberCountRequirement))
+		{
+			sender->Message(Chat::Red, "You are unable to enter an instance because you are not a part of a raid with %i players at or above level %i present total.",
+				RuleI(Quarm, AutomatedRaidRotationRaidNonMemberCountRequirement),
+				RuleI(Quarm, AutomatedRaidRotationRaidGuildLevelRequirement));
+			return false;
+		}
+
+		if (player_raid->GetRaidLeaderCharacterID() != sender->CharacterID())
+		{
+			uint32 leader_instanceid = zone->GetZoneInstanceIDByCharacterAndZone(player_raid->GetRaidLeaderCharacterID(), zoneid);
+			if (ourZoneInstanceID == GUILD_NONE && leader_instanceid != GUILD_NONE)
```

Comment:
Isn't ourZoneInstanceID  always GUILD_NONE at this part of the zone? it's only set in the if statement block above